config/terraform_cloud_workspace.go

```diff
@@ -0,0 +1,129 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/consul-terraform-sync/logging"
+)
+
+// TerraformCloudWorkspaceConfig is an enterprise-only configuration that controls
+// workspace attributes that are specific to a task.
+type TerraformCloudWorkspaceConfig struct {
+	ExecutionMode *string `mapstructure:"execution_mode"`
+	AgentPoolID   *string `mapstructure:"agent_pool_id"`
+	AgentPoolName *string `mapstructure:"agent_pool_name"`
+}
+
+// Copy returns a deep copy of this configuration.
+func (c *TerraformCloudWorkspaceConfig) Copy() *TerraformCloudWorkspaceConfig {
+	if c == nil {
+		return nil
+	}
+
+	var o TerraformCloudWorkspaceConfig
+	o.ExecutionMode = c.ExecutionMode
+	o.AgentPoolID = c.AgentPoolID
+	o.AgentPoolName = c.AgentPoolName
+
+	return &o
+}
+
+// Merge combines all values in this configuration with the values in the other
+// configuration, with values in the other configuration taking precedence.
+func (c *TerraformCloudWorkspaceConfig) Merge(o *TerraformCloudWorkspaceConfig) *TerraformCloudWorkspaceConfig {
+	if c == nil {
+		if o == nil {
+			return o
+		}
+		return o.Copy()
+	}
+
+	if o == nil {
+		return c.Copy()
+	}
+
+	r := c.Copy()
+
+	if o.ExecutionMode != nil {
+		r.ExecutionMode = StringCopy(o.ExecutionMode)
+	}
+
+	if o.AgentPoolID != nil {
+		r.AgentPoolID = StringCopy(o.AgentPoolID)
+	}
+
+	if o.AgentPoolName != nil {
+		r.AgentPoolName = StringCopy(o.AgentPoolName)
+	}
+
+	return r
+}
+
+// Finalize ensures that the receiver contains no nil pointers.
+func (c *TerraformCloudWorkspaceConfig) Finalize() {
+	if c == nil { // config is not required, return early
+		return
+	}
+
+	if c.ExecutionMode == nil {
+		c.ExecutionMode = String("remote")
```

Comment:
can we add this string as a constant? "DefaultExecutionMode" or something to that effect?

Reply:
you got rid of this in the next commit, so all good